02_bubble/paraganz/main.go

```diff
@@ -0,0 +1,27 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubble(s []int) []int {
+	cnt := len(s)
+	for i := 0; i < cnt; i++ {
+		for j := i + 1; j < cnt; j++ {
+			if s[i] > s[j] {
+				tmp := s[i]
+				s[i], s[j] = s[j], tmp
+				j--
+			}
+		}
+	}
+	return s
```

Comment:
Can't mutate the original. Make a copy. But first write a test that detects the lack of copying.

Reply:
Done